kinesis/kinesis.go

```diff
@@ -75,26 +77,28 @@ type OutputPlugin struct {
 	// Partition key decides in which shard of your stream the data belongs to
 	partitionKey string
 	// Decides whether to append a newline after each data record
-	appendNewline                bool
-	timeKey                      string
-	fmtStrftime                  *strftime.Strftime
-	lastInvalidPartitionKeyIndex int
-	client                       PutRecordsClient
-	records                      []*kinesis.PutRecordsRequestEntry
-	dataLength                   int
-	timer                        *plugins.Timeout
-	PluginID                     int
-	random                       *random
+	appendNewline bool
+	timeKey       string
+	fmtStrftime   *strftime.Strftime
+	client        PutRecordsClient
+	timer         *plugins.Timeout
+	PluginID      int
+	random        *random
+	Concurrency   int
+	retryLimit    int
```

Comment:
nit: If I am not wrong, this is `concurrency_retries`. Can we rename it to `concurrencyRetryLimit` to be consistent.